### PR TITLE
MM-46443 : Gekidou - removes save button on settings screens

### PR DESCRIPTION
**Pull request by avinashlng1080**
#### Summary
This PR  removes all the save button on the Settings screen.  The new value are now sent over to the server on navigating back.

To the reviewers, kindly  test it on Android to double check.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-46443

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/screens/settings/config.ts [hunk 1]
```diff
@@ -16,6 +17,18 @@ export const getSaveButton = (buttonId: string, intl: IntlShape, color: string)
     ...typography('Body', 100, 'SemiBold'),
 });
 
+export const gotoSettingsScreen = (screen: string, title: string) => {
+    const passProps = {};
+    const options = {
+        topBar: {
+            backButton: {
+                popStackOnPress: false,
```
Comment (enahum) [inline app/screens/settings/config.ts:25]: why are we doing this?

File: app/screens/settings/display_timezone_select/index.tsx [hunk 2]
```diff
@@ -80,7 +77,8 @@ const SelectTimezones = ({componentId, onBack, currentTimezone}: SelectTimezones
     const [timezones, setTimezones] = useState<string[]>(EMPTY_TIMEZONES);
     const [initialScrollIndex, setInitialScrollIndex] = useState<number|undefined>();
     const [searchRegion, setSearchRegion] = useState<string|undefined>(undefined);
-    const [manualTimezone, setManualTimezone] = useState(currentTimezone);
+
+    // const [manualTimezone, setManualTimezone] = useState(currentTimezone);
```
Comment (jfrerich) [inline app/screens/settings/display_timezone_select/index.tsx:81]: Remove if not going to be used.

**Review by jfrerich — COMMENTED** [review]
image

**Comment by avinashlng1080** [comment]
Awaiting PR #6584

**Comment by josephbaylon** [comment]
> Awaiting PR #6584

@avinashlng1080 PR has been merged

File: app/screens/settings/display_timezone_select/index.tsx (hunk 2 above)
Comment (avinashlng1080) [reply app/screens/settings/display_timezone_select/index.tsx:81]: corrected

File: app/screens/settings/display_clock/display_clock.tsx [hunk 3]
```diff
@@ -23,51 +21,41 @@ const CLOCK_TYPE = {
...
+        if (canSave) {
+            const timePreference: PreferenceType = {
+                category: Preferences.CATEGORY_DISPLAY_SETTINGS,
+                name: 'use_military_time',
+                user_id: currentUserId,
+                value: `${isMilitaryTimeFormat}`,
+            };
+
+            savePreference(serverUrl, [timePreference]);
+        }
```
[the remark this answers is not in the document]
Comment (avinashlng1080) [reply app/screens/settings/display_clock/display_clock.tsx:51]: I think it's the same - one is returning earlier but accomplishing the same.

File: app/screens/settings/config.ts (hunk 1 above)
Comment (avinashlng1080) [reply app/screens/settings/config.ts:25]: To hijack the software back button and run a callback that will save the newly changed value for each setting - https://wix.github.io/react-native-navigation/docs/customBackNavigation#handling-the-software-back-button

**Comment by josephbaylon** [comment]
@avinashlng1080 `Mentions`(CRT) or `Mentions and Replies`(non-CRT) > Your case sensitive first name (toggle) change is not getting saved on either iOS or Android

File: app/screens/settings/notification_auto_responder/notification_auto_responder.tsx [hunk 4]
```diff
@@ -66,46 +64,37 @@ const NotificationAutoResponder = ({currentUser, componentId}: NotificationAutoR
...
-            },
-        });
-        fetchStatusInBatch(serverUrl, currentUser.id);
+        const canSaveSetting = initialAutoResponderActive !== autoResponderActive || initialOOOMsg !== autoResponderMessage;
+
+        if (canSaveSetting) {
+            updateMe(serverUrl, {
+                notify_props: {
+                    ...notifyProps,
+                    auto_responder_active: `${autoResponderActive}`,
```
Comment (enahum) [inline app/screens/settings/notification_auto_responder/notification_auto_responder.tsx:86]: what is the value set when the field is empty?

File: app/screens/settings/notification_email/notification_email.tsx [hunk 5]
```diff
@@ -81,47 +77,39 @@ const NotificationEmail = ({componentId, currentUser, emailInterval, enableEmail
...
-            promises.push(savePrefPromise);
+    const saveEmail = useCallback(() => {
+        const canSaveSetting = notifyInterval !== initialInterval || emailThreads !== initialEmailThreads;
+        if (canSaveSetting) {
+            const promises = [];
+            const updatePromise = updateMe(serverUrl, {
+                notify_props: {
+                    ...notifyProps,
+                    email: `${sendEmailNotifications && notifyInterval !== Preferences.INTERVAL_NEVER.toString()}`,
+                    email_threads: emailThreads ? 'all' : 'mention',
```
Comment (enahum) [inline app/screens/settings/notification_email/notification_email.tsx:90]: does this have to be set for all crt and non-crt?

File: app/screens/settings/notification_push/notification_push.tsx [hunk 6]
```diff
@@ -40,48 +35,31 @@ const NotificationPush = ({componentId, currentUser, isCRTEnabled, sendPushNotif
...
+        return p || pT || pS;
+    }, [notifyProps, pushSend, pushStatus, pushThread]);
 
     const saveNotificationSettings = useCallback(() => {
-        const notify_props = {...notifyProps, push: pushSend, push_status: pushStatus, push_threads: pushThread};
-        updateMe(serverUrl, {notify_props} as unknown as UserNotifyProps);
+        const canSave = canSaveSettings();
+        if (canSave) {
+            const notify_props = {...notifyProps, push: pushSend, push_status: pushStatus, push_threads: pushThread};
+            updateMe(serverUrl, {notify_props} as unknown as UserNotifyProps);
```
Comment (enahum) [inline app/screens/settings/notification_push/notification_push.tsx:55]: why unknown? isn't there a better way? like `const notify_pros: UserNotifyProps = {...`

File: app/screens/settings/notification_auto_responder/notification_auto_responder.tsx (hunk 4 above)
Comment (avinashlng1080) [reply app/screens/settings/notification_auto_responder/notification_auto_responder.tsx:86]: `"false"`

**Comment by avinashlng1080** [comment]
> @avinashlng1080 `Mentions`(CRT) or `Mentions and Replies`(non-CRT) > Your case sensitive first name (toggle) change is not getting saved on either iOS or Android

@josephbaylon  can you please try again ?

File: app/screens/settings/notification_push/notification_push.tsx (hunk 6 above)
Comment (avinashlng1080) [reply app/screens/settings/notification_push/notification_push.tsx]: fixed

**Review by enahum — APPROVED** [review]

**Review by jfrerich — APPROVED** [review]
LGMT after checking your commit that fixed my issue.

**Review by matthewbirtch — APPROVED** [review]
Looks good to me!

**Comment by josephbaylon** [comment]
> > @avinashlng1080 `Mentions`(CRT) or `Mentions and Replies`(non-CRT) > Your case sensitive first name (toggle) change is not getting saved on either iOS or Android
> 
> @josephbaylon can you please try again ?

@avinashlng1080 The issue is now fixed. However with the latest change on email notification for CRT mode, when user transitions from `Never` to `Immediately` without tapping on back navigation yet, the toggle for `Threads reply notifications` is not shown right away. It is only shown when user navigates back and then goes back into email notifications. This is inconsistent with how the toggle `Thread replies` appears right away on `Push Notifications` when user transitions to `Mentions, direct messages only` without tapping on back navigation yet.

**Comment by enahum** [comment]
@josephbaylon please do re-review again, I've addressed your feedback

**Review by josephbaylon — APPROVED** [review]
LGTM.
1. On both iOS and Android (CRT and non-CRT), save buttons are now removed and any changes on the notification settings are only saved when user taps on app navigation back
2. On both iOS and Android (CRT and non-CRT), any changes on the notifications settings are not saved when user closes/re-opens app without tapping on app navigation back
3. On Android (CRT and non-CRT), any changes on the notification settings are not saved when user uses hardware back button instead of app navigation back

**Pull request merged**